### ISSUE TITLE
Build of 1.8.7 fails on musl

### ISSUE THREAD

**Issue by Cogitri** [gnunn1/tilix#1624]
Hello,

since 1.8.7 Tilix fails to build:

```
...
rce/secretc/secrettypes.d source/x11/X.d source/x11/Xlib.d -oftilix.o
source/gx/tilix/terminal/terminal.d(2839): Error: module `core.sys.posix.stdlib` import `grantpt` not found
source/gx/tilix/terminal/terminal.d(2839): Error: module `core.sys.posix.stdlib` import `unlockpt` not found
source/gx/tilix/terminal/terminal.d(2839): Error: module `core.sys.posix.stdlib` import `ptsname` not found
make[1]: *** [Makefile:1037: tilix.o] Error 1
make[1]: Leaving directory '/builddir/tilix-1.8.7'
```

See https://travis-ci.org/void-linux/void-packages/jobs/475338660 for a full build log

**Comment by gnunn1** [comment]
Unfortunately I am not familiar with musl since none of my container work involves alpine and I have no plans to start officially supporting or testing tilix with musl. 

Having said that, the code that seems to be causing the issue was added to support flatpak, it would be possible to add a switch to compile tilix without flatpak support. I can look at adding this at some point but it's not something that would be a priority for me, happy to accept a PR for it though.

**Comment by Cogitri** [comment]
Alright, thanks for answering so quickly. I'll see what I can do with my (very) limited D knowledge :)

**Comment by gnunn1** [comment]
You can try asking on the dlang forums, I know there was some work done around musl previously and the D guys are pretty friendly and helpful.

https://forum.dlang.org/group/general

**Comment by Cogitri** [comment]
Yeah. 1.8.5 worked just fine, so I guess this should be easy to fix :)

**Comment by Cogitri** [comment]
Alright, as can be seem https://github.com/dlang/druntime/blob/a59c3832fc00ab127018d9a570ba03520437524f/src/core/sys/posix/stdlib.d#L298

It seems like grantpt can't be used on musl. I'll just try to patch it back to 1.8.5 behavior for musl for our purposes.

**Comment by Cogitri** [comment]
We've patched it locally for now, we'll just keep that patch downstream. Would be great if you could fix it properly somewhere down the line though!
